sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverAsyncClient.java

```diff
@@ -1200,22 +1201,27 @@ private String getLinkName(String sessionId) {
         } else if (!receiverOptions.isSessionReceiver()) {
             return monoError(logger, new IllegalStateException("Cannot get session state on a non-session receiver."));
         }
+
+        Mono<byte[]> result;
+
         if (sessionManager != null) {
-            return sessionManager.getSessionState(sessionId);
+            result =  sessionManager.getSessionState(sessionId);
         } else {
-            return connectionProcessor
+            result =  connectionProcessor
                 .flatMap(connection -> connection.getManagementNode(entityPath, entityType))
                 .flatMap(channel -> channel.getSessionState(sessionId, getLinkName(sessionId)));
         }
+
+        return result.onErrorMap((err) -> mapError(err, ServiceBusErrorSource.RECEIVE));
     }
 
     /**
-     * Map the error to {@link ServiceBusReceiverException}
+     * Map the error to {@link ServiceBusException}
      */
     private Throwable mapError(Throwable throwable, ServiceBusErrorSource errorSource) {
         // If it is already `ServiceBusReceiverException`, we can just throw it.
-        if (!(throwable instanceof ServiceBusReceiverException)) {
-            return new ServiceBusReceiverException(throwable, errorSource);
+        if (!(throwable instanceof ServiceBusException)) {
```

Comment:
nit: update the comment too.

Reply:
Removed the comment - realized what the function was doing was pretty obvious.